### ISSUE TITLE
Mix.Appsignal.Helper.ensure_downloaded/1 does not overwrite old agent versions

### ISSUE THREAD

**Issue by Iwark** [appsignal/appsignal-elixir#138]
After upgrading appsignal to 0.12.1, I got an error as starting my phoenix server in production(Amazon Linux): 

```
[debug] Appsignal starting.
[error] Error loading NIF, Appsignal integration disabled!

Failed to load NIF library: '/home/ec2-user/myapp/_build/prod/lib/appsignal/priv/appsignal_extension.so: undefined symbol: appsignal_running_in_container'

Is your operating system (x86_64-unknown-linux-gnu) supported?
Please check http://docs.appsignal.com/support/operating-systems.html
```

It works fine when appsignal's version is 0.11.6

Is it a bug or Am I doing something wrong?

**Comment by jeffkreeftmeijer** [comment]
Hey @Iwark, 

Thanks for reporting this! We're having some problems downloading new versions of the agent when an older version is already installed. We'll work on this to make sure upgrades are a little smoother in the future.

In the meantime; could you try removing your `_build` directory and trying again? Please let me know if that worked for you.

**Comment by Iwark** [comment]
Hi @jeffkreeftmeijer,

Thank you! I removed `_build` directory and it worked like a charm :)

**Comment by jeffkreeftmeijer** [comment]
@Iwark that's great to hear! 🎉 

We're keeping this issue open to track our progress on improving the update process and removing the need to manually delete the `_build` directory.

**Comment by tombruijn** [comment]
This was fixed in 0.12.2. It should no longer happen now.

The problem was that when upgrading the AppSignal package the c-extension of the build was not reset/cleared. Our check to see if the agent needs to be downloaded assumed we the newer version would be installed in a new dependency directory. Since this is not the case we added an extra check to see if a new version should be downloaded.

We've added a `appsignal.version` file to the downloaded agent package so we know what version is currently installed locally and if we need to upgrade it. This will also work upgrading from versions lower than 0.12.2 to version 0.12.2 and higher. These agent packages have no version file and we will assume the agent version needs to be upgraded then.
